android-core/src/androidTest/kotlin/com.mparticle/networking/MParticleBaseClientImplTest.kt

```diff
@@ -20,11 +21,25 @@ class MParticleBaseClientImplTest : BaseCleanInstallEachTest() {
         startMParticle(MParticleOptions.builder(mContext).credentials(apiKey, "secret"))
         val baseClientImpl = AccessUtils.getApiClient() as MParticleBaseClientImpl
         for (endpoint in MParticleBaseClientImpl.Endpoint.values()) {
-            defaultUrls[endpoint] = baseClientImpl.getUrl(endpoint, endpoint.name)
+            defaultUrls[endpoint] = baseClientImpl.getUrl(endpoint, endpoint.name, false)
         }
         MParticle.setInstance(null)
     }
 
+    @Test
```

Comment:
We should add more comprehensive testing. Looks like this only tests the final URL creation, not ensuring it pulls the correct prefix from different keys, and doesn't test all pods or keys without prefixes.

See the iOS tests here for reference: https://github.com/mParticle/mparticle-apple-sdk/blob/14168f9167f15f8b850b93ea1004273cf927bd0e/UnitTests/MPNetworkCommunicationTests.m#L841

Web also tests similarly to iOS.